### ISSUE TITLE
Migrate live regions to KDS useKLiveRegion

### ISSUE THREAD

**Issue by MisRob** [learningequality/kolibri#12326]
## Blocked by

https://github.com/learningequality/kolibri-design-system/issues/668

## Summary

Refactor Kolibri to use `useKLiveRegion` for reasons mentioned in https://github.com/learningequality/kolibri-design-system/issues/668.

Current live regions in Kolibri:

- (1) `aria-live`
    - [search results](https://github.com/search?q=repo%3Alearningequality%2Fkolibri+aria-live&type=code)
- (2) `role="alert"`
    - [search results](https://github.com/search?q=repo%3Alearningequality%2Fkolibri+role%3D%22alert%22&type=code)
    - roughly corresponds to `assertive` `aria-live` but with added semantic meaning as on some screen readers it prepends announcements with *“Alert: …”*
- (3) `role="timer"`
    - [search results](https://github.com/search?q=repo%3Alearningequality%2Fkolibri+role%3D%22timer%22&type=code)

**Comment by MisRob** [comment]
## To decide/discuss

- Should `<KLiveRegion>` also support the timer role (3) or is it best to leave that one extra live region as is?
- Is there a place where we could put `<KLiveRegion>` in Kolibri so that it takes effect for all plugins, or should each plugin include it?

**Comment by MisRob** [comment]
Decisions:
- Won't add timer role. Reason: There's just one use-case. Would be good to QA how it plays together with the global live region though.

**Comment by MisRob** [comment]
Reported by @radinamatic in https://github.com/learningequality/kolibri/pull/12475

![Screenshot from 2024-08-07 17-34-08](https://github.com/user-attachments/assets/e976e605-4c6c-4a92-bbc2-a990312701a1)

**Comment by lokesh-sagi125** [comment]
hey @MisRob  can i work on this issue? , i see that the blocking issue has been resolved

**Comment by MisRob** [comment]
Hi @lokesh-sagi125,

I'd be fine assigning to you since you already have some nice experience with our codebases, however note even though `useKLiveRegion` already takes care of many things, this is a bit more complex issue because you'd likely need to refactor some parts of global components such as snackbars. For that, you would need to understand how it's used in various places.

Also, I am not sure if the current live regions work well, and I can't prepare expected behavior requirements myself right now.  So you may need to do some study around live regions and then propose yourself how a particular live region should behave, and test refactored places with a screen reader thoroughly. 

You could work one by one, and one way I can support here would be to briefly confirm the overall direction for each case.

So perhaps take some time to think whether it's something you'd like to tackle :) Let us know any time and thanks!

**Comment by lokesh-sagi125** [comment]
yes, @MisRob, I've worked on issues involving multiple components and have learned how to test them individually. I think I am ready to tackle this problem and would need some guidance, which I'm sure you'll provide me with :).

**Comment by lokesh-sagi125** [comment]
i will go through the documentation and the blocking issue to better understand the issue.

**Comment by MisRob** [comment]
@lokesh-sagi125 alright yes I think it's reasonable to spend some time at first familiarizing yourself with everything. I won't be able to provide much guidance except of what I already mentioned and some high-level confirmation of your own proposal, that's why it's rather an advanced task - it requires your own research. Let me assign you just to make sure it's reserved for you since you're spending time on it with the study, but no problem to unassign if you change your mind. Just message us.

**Comment by lokesh-sagi125** [comment]
thank you @MisRob

**Comment by lokesh-sagi125** [comment]
Hi, I have been unassigned, was inactive due to exams, and gsoc proposal , i will get back to the issue once i have submitted the proposal

**Comment by MisRob** [comment]
HI @lokesh-sagi125, no problem, I will re-assign you. If you get unassigned again in a month, just ping me here.
